### ISSUE TITLE
Need an easy way to return the value of a single task in `par`

### ISSUE THREAD

**Issue by cpettitt** [linkedin/parseq#25]
We have a use case where we have 3 tasks:

- t1 produces some value
- t2 is dependent on t1 and produces a value
- t3 is dependent on t1, but does not produce a value

We'd like our composite task to return the result of t2. We could sequence them like:

```java
Tasks.seq(t1, t3, t2)
```

However, t2 and t3 have no inter-dependency, so we'd sacrifice parallelism unnecessarily. Ideally, we should be able to parallelize these tasks without requiring a custom task. I propose the following way of composing these tasks:

```java
Tasks.seq(t1, Tasks.par(t2, t3), Tasks.value(t2))
```

This requires that we add `Tasks.value` as a new primitive in ParSeq. It effectively acts as a hint about the return value: when it is run it just returns the value of another task.

**Comment by jodzga** [comment]
This is solved by withSideEffect in parseq 2.x:
t1.withSideEffect(t2).andThen(t2)
